mm2src/mm2_p2p/src/behaviours/mod.rs

```diff
@@ -0,0 +1,460 @@
+                    return;
+                }
+
+                panic!("Request received more than 3 times");
+            }
+        }
+
+        let request_handler = Arc::new(Mutex::new(RequestHandler::default()));
+
+        let mut receivers = Vec::new();
+        for _ in 0..3 {
+            let handler = request_handler.clone();
+            let receiver_port = next_port();
+            let receiver = Node::spawn(receiver_port, vec![], move |cmd_tx, event| {
+                let mut handler = handler.lock().unwrap();
+                handler.handle(cmd_tx, event)
+            })
+            .await;
+            receivers.push((receiver_port, receiver));
+        }
+
+        let mut sender = Node::spawn(
+            next_port(),
+            receivers.iter().map(|(port, _)| *port).collect(),
+            |_, _| (),
+        )
+        .await;
+
+        sender.wait_peers(3).await;
+
+        let (response_tx, response_rx) = oneshot::channel();
+        sender
+            .send_cmd(AdexBehaviourCmd::RequestAnyRelay {
+                req: b"test request".to_vec(),
+                response_tx,
+            })
+            .await;
+
+        let (_peer_id, res) = response_rx.await.unwrap().unwrap();
+        assert_eq!(res, b"success 3 request".to_vec());
+    }
+
+    #[tokio::test]
+    async fn test_request_response_none() {
+        let _ = env_logger::try_init();
+
+        let request_received = Arc::new(AtomicBool::new(false));
+        let request_received_cpy = request_received.clone();
+
+        let node1_port = next_port();
+        let _node1 = Node::spawn(node1_port, vec![], move |mut cmd_tx, event| {
+            let (request, response_channel) = match event {
+                AdexBehaviourEvent::RequestResponse(RequestResponseBehaviourEvent::InboundRequest {
+                    request,
+                    response_channel,
+                    ..
+                }) => (request.req, AdexResponseChannel(response_channel)),
+                _ => return,
+            };
+
+            request_received_cpy.store(true, Ordering::Relaxed);
+            assert_eq!(request, b"test request");
+
+            let res = AdexResponse::None;
+            cmd_tx
+                .try_send(AdexBehaviourCmd::SendResponse { res, response_channel })
+                .unwrap();
+        })
+        .await;
+
+        let mut node2 = Node::spawn(next_port(), vec![node1_port], |_, _| ()).await;
+
+        node2.wait_peers(1).await;
+
+        let (response_tx, response_rx) = oneshot::channel();
+        node2
+            .send_cmd(AdexBehaviourCmd::RequestAnyRelay {
+                req: b"test request".to_vec(),
+                response_tx,
+            })
+            .await;
+
+        assert_eq!(response_rx.await.unwrap(), None);
+        assert!(request_received.load(Ordering::Relaxed));
+    }
+
+    #[tokio::test]
+    #[cfg(target_os = "linux")] // https://github.com/KomodoPlatform/atomicDEX-API/issues/1712
```

Comment:
Why does this test frequently fail on Windows? Can the reason be fixed instead of running it only on specific OS?

Reply:
This change was made a while ago when I was reviewing the entire tests in mm2. Because I didn't had enough time, I didn't check into the reasons behind the failures. I believe it would be more appropriate to address this in a separate task, referring to issue mentioned in the comment.